en/peripherals/camera.md

```diff
@@ -33,17 +33,19 @@ Mode | Description
 
 ## Trigger Hardware Configuration {#hardware_setup}
 
-You can choose which pins to use for triggering using the [TRIG_PINS](../advanced_config/parameter_reference.md#TRIG_PINS) parameter.
+On Pixhawk FMUv5x boards use the board-specific camera capture pin (PI0) to signal PX4 that an image has been captured (this is not available on earlier Pixhawk FMU versions).
```

Comment:
@bkueng Updated this to make it clear what capture is, and (my assumption) that camera capture indication is only supported on FMUv5. I know this was planned to also be indicated in MAVLink but I don't think Daniele ever finished the implemetation - correct?

Reply:
Other boards use the FMU PWM pin 6 (hardcoded).
From the code, the `CAMERA_TRIGGER` mavlink message is sent out as well, so should be all there.

Reply:
@bkueng OK, I updated the capture info for FMU PWM pin 6. 

> the CAMERA_TRIGGER mavlink message is sent out as well, so should be all there.

Yes, but we're talking about camera *capture*, not trigger. That message should be sent by the camera and processed by PX4 (not "sent"). I have a feeling that we don't yet process that signal.

Reply:
See https://github.com/PX4/Firmware/blob/master/src/drivers/camera_capture/camera_capture.cpp#L159. I'm not sure how this is supposed to work, but that's how it is currently.
There's also a `CAMERA_CAPTURE` message. Quite confusing.

Reply:
This was discussed in https://github.com/PX4/Firmware/pull/10295 but eventually merged in https://github.com/PX4/Firmware/pull/11395 (for some reason I though it got abandoned). Anyway the theory here is that you publish the trigger message **after** you get the capture (from a hotshoe say), and this gives you a better timestamp for geotagging.

But I really don't get this. I mean I understand how it will work for a camera  attached to the FC, but if you have a mavlink camera then the CAMERA_TRIGGER is what sets off triggering, so this becomes recursive. 

Perhaps @DanielePettenuzzo can explain, but I think the mavlink case is an omission/enhancement that should not block this right?

Reply:
> But I really don't get this. I mean I understand how it will work for a camera attached to the FC, but if you have a mavlink camera then the CAMERA_TRIGGER is what sets off triggering, so this becomes recursive.

Exactly, my main point of confusion as well.

> OK, I updated the capture info for FMU PWM pin 6.

I don't see this?

Reply:
>> But I really don't get this. I mean I understand how it will work for a camera attached to the FC, but if you have a mavlink camera then the CAMERA_TRIGGER is what sets off triggering, so this becomes recursive.

> Exactly, my main point of confusion as well.

Arrgg, I got this wrong, and am sorry to have confused you/us. [CAMERA_TRIGGER](https://mavlink.io/en/messages/common.html#CAMERA_TRIGGER) is always a notification. It is emitted by either the camera driver or the camera capture driver. It will also be emitted by a MAVLink camera. 

If you use a camera capture there will be multiples of this flying around, with the same sequence number and different timestamp. So essentially you'll have some component that is smart enough to resolve this, as described in https://github.com/PX4/Firmware/pull/10295#issuecomment-425932056